src/main/java/com/yana/internship/service/ApartmentService.java

```diff
@@ -0,0 +1,59 @@
+package com.yana.internship.service;
+
+import com.yana.internship.entity.Apartment;
+import com.yana.internship.entity.Country;
+import com.yana.internship.repository.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@Transactional
+public class ApartmentService {
+    private static final Logger logger = LoggerFactory.getLogger(ApartmentService.class);
+    private final ApartmentRepository apartmentRepository;
+    private final TariffRepository tariffRepository;
+    private final AddressRepository addressRepository;
+    private final ImageRepository imageRepository;
+
+    @Autowired
+    public ApartmentService(ApartmentRepository apartmentRepository, TariffRepository tariffRepository, AddressRepository addressRepository, ImageRepository imageRepository) {
+        this.apartmentRepository = apartmentRepository;
+        this.tariffRepository = tariffRepository;
+        this.addressRepository = addressRepository;
+        this.imageRepository = imageRepository;
+    }
+
+    public Apartment create(Apartment apartment) {
+        tariffRepository.save(apartment.getTariff());
+        addressRepository.save(apartment.getAddress());
+        imageRepository.saveAll(apartment.getImages());
+        logger.info("Create apartment with id {}", apartment.getId());
```

Comment:
I think the log have wrong place...
Please move it to beginning of the method

And... do we know apartment Id before creation?